crates/runtime/src/lib.rs

```diff
@@ -155,24 +155,35 @@ impl Runtime {
         if let Some(app) = app_lock.as_ref() {
             for ds in &app.datasets {
                 status::update_dataset(ds.name.clone(), status::ComponentStatus::Initializing);
-                self.load_dataset(ds);
+                self.load_dataset(ds, &app.datasets);
             }
         }
     }
 
     // Caller must set `status::update_dataset(...` before calling `load_dataset`. This function will set error/ready statues appropriately.`
-    pub fn load_dataset(&self, ds: &Dataset) {
+    pub fn load_dataset(&self, ds: &Dataset, all_datasets: &[Dataset]) {
         let df = Arc::clone(&self.df);
         let spaced_tracer = Arc::clone(&self.spaced_tracer);
         let shared_secrets_provider: Arc<RwLock<secrets::SecretsProvider>> =
             Arc::clone(&self.secrets_provider);
 
         let ds = ds.clone();
 
+        let existing_tables = all_datasets
+            .iter()
+            .map(|d| d.name.clone())
+            .collect::<Vec<String>>();
+
         tokio::spawn(async move {
             loop {
                 let secrets_provider = shared_secrets_provider.read().await;
 
+                if !verify_dependent_tables(&ds, &existing_tables, df.clone()).await {
```

Comment:
```suggestion
                if !verify_dependent_tables(&ds, &existing_tables, Arc::clone(&df)).await {
```

I find this syntax clearer to understand - we're just cloning a pointer, not the object.